rasa/shared/nlu/training_data/training_data.py

```diff
@@ -755,6 +759,73 @@ def __init__(
             training_examples, entity_synonyms, regex_features, lookup_tables, responses
         )
 
+    @staticmethod
+    def _bytes_feature(array: np.ndarray) -> tf.train.Feature:
+        value = tf.io.serialize_tensor(array)
+        if isinstance(value, type(tf.constant(0))):
+            value = value.numpy()
+        return tf.train.Feature(bytes_list=tf.train.BytesList(value=[value]))
+
+    @staticmethod
+    def _int_feature(array: Union[Tuple[int], np.ndarray]) -> tf.train.Feature:
+        return tf.train.Feature(int64_list=tf.train.Int64List(value=array))
+
+    @staticmethod
+    def _construct_key(
+        attribute: Text, feature_type: Text, origin: Text, is_dense: bool
+    ) -> Text:
+        prefix = (
+            f"{attribute}{TFRECORD_KEY_SEPARATOR}{feature_type}"
+            f"{TFRECORD_KEY_SEPARATOR}{origin}{TFRECORD_KEY_SEPARATOR}"
+        )
+
+        if is_dense:
+            return f"{prefix}dense"
+        return f"{prefix}sparse"
+
+    @staticmethod
+    def _deconstruct_key(key: Text) -> Tuple[Text, Text, Text, bool, Text]:
```

Comment:
Rename the function to `_deconstruct_tf_record_key` or similar. `_deconstruct_key` is slightly ambiguous.